src/nupic/algorithms/Cells4.cpp

```diff
@@ -1036,11 +1029,12 @@ void Cells4::updateInferenceState(const std::vector<UInt> & activeColumns)
   _infActiveStateT1 = _infActiveStateT;
   _infPredictedStateT1 = _infPredictedStateT;
   memcpy(_cellConfidenceT1, _cellConfidenceT, _nCells * sizeof(_cellConfidenceT[0]));
+  memset(_cellConfidenceT, 0, _nCells * sizeof(_cellConfidenceT[0]));
```

Comment:
I tried this in f12abc0 which resulted in memleaks and crashes, so I reverted it. Now retrying with 
Anyway, the time is: for-loops >> memset > only-1 memset, at the cost we are not reseting to `0` where the comments say so (to be honest, comments said, but code did otherwise in the orig version :P )